lms/resources/lti_launch.py

```diff
@@ -38,31 +38,14 @@ def get_or_create_course(self):
         tool_consumer_instance_guid = params["tool_consumer_instance_guid"]
         context_id = params["context_id"]
 
-        authority_provided_id = course_service.generate_authority_provided_id(
-            tool_consumer_instance_guid, context_id
-        )
-
-        legacy_course = course_service.get_or_create(authority_provided_id)
-        # Capture the state here, before any other SQLA query does an implicit flush and then
-        # removing legacy_course from `db.new`
-        is_new_legacy_course = legacy_course in self._request.db.new
-
-        course = course_service.upsert(
-            authority_provided_id,
+        return course_service.upsert(
+            course_service.generate_authority_provided_id(
+                tool_consumer_instance_guid, context_id
+            ),
```

Comment:
I think this refactor has tidied this up to the extent it's pointing out there's something weird about `generate_authority_provided_id`. As far as I can tell it's never called for anything other than just to feed-back to the course service.

This makes me think it's not really an external method. The two patterns I can spot are:

```
    course_service.upsert(
        course_service.generate_authority_provided_id(
            tool_consumer_instance_guid, context_id
        ),
        ...
    )
```

```
    course_service.get(
        course_service.generate_authority_provided_id(
            self.tool_consumer_instance_guid, course_id
        )
    )
```

I can't spot `upsert` or `get` ever being called by anything else, which makes me think this should be an internal method and exposed through the interface.